src/nvim/decoration_defs.h

```diff
@@ -52,9 +53,10 @@ typedef struct {
   DecorPriority priority;
   int hl_id;
   schar_T conceal_char;
+  String url;
```

Comment:
As these need allocation anyway, there is no point in having them in  DecorHighlightInline (as you note, these be converted to `ext` anyway so it is just waste of space here). better to use a code path more like `line_hl_id` etc to construct them.

Reply:
Done in a90c842737f3bd38c4aafab09da7d3fb0182b1af

Not sure how best to handle ephemeral extmarks with URLs now. When they are part of `DecorHighlightInline` they get converted in `decor_sh_from_inline`. I explicitly am adding a new DecorSignHighlight when `sign.url.data` is non null for ephemeral extmarks, which works, but maybe there is a better way.